SPEC.md

```diff
@@ -1352,13 +1352,55 @@ Loops are distinct from scatter clauses because the body of a while loop needs t
 
 ### Conditionals
 
-:pig2: Coming soon in [Cromwell](https://github.com/broadinstitute/cromwell)
+:pig2: Available in [Cromwell](https://github.com/broadinstitute/cromwell) version 24 and higher
 
 ```
 $conditional = 'if' '(' $expression ')' '{' $workflow_element* '}'
 ```
 
-Conditionals only execute the body if the expression evaluates to true
+Conditionals only execute the body if the expression evaluates to true.
+
+* When a call's output is referenced outside the same containing `if` it will need to be handled as an optional type. E.g.
+```
+workflow foo {
+  # Call 'x', producing a Boolean output:
+  call x
+  Boolean x_out = x.out
+
+  # Call 'y', producing an Int output, in a conditional block:
+  if (x_out) {
+    call y
+    Int y_out = y.out
+  }
+  
+  # Outside the if block, we have to handle this output as optional:
+  Int? y_out_maybe = y.out
+
+  # Call 'z' which takes an optional Int input:
+  call z { input: optional_int = y_out_maybe }
+}
+```
+* Optional types can be coalesced by using the `select_all` and `select_first` array functions:
+```
+workflow foo {
+  Array[Int] scatter_range = [1, 2, 3, 4, 5]
+  scatter (i in scatter_range) {
+    call x { input: i = i }
+    if (x.validOutput) {
+      Int x_out = x.out
+    }
+  }
+
+  # Because it was declared inside the scatter and the if-block, the type of x_out is different here:
+  Array[Int?] x_out_maybes = x_out
+
+  # We can select only the valid elements with select_all:
+  Array[Int] x_out_valids = select_all(x_out_maybes)
```

Comment:
Oh wait, actually I think it is slightly different that what you're expecting. 

If the `x.validOutput` is true, then that entry in `x_out_maybes` is a value. If `x.validOutput` is false, there is still an entry in `x_out_maybes` for that, but it's an empty optional value. So this `select_all` is necessary after all to filter out the empties. Does that make sense?